cmdtests/gitbase_back_comp_test.go

```diff
@@ -0,0 +1,274 @@
+
+	cmdtest "github.com/src-d/engine/cmdtests"
+
+	"gopkg.in/src-d/regression-core.v0"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/stretchr/testify/suite"
+)
+
+var prevVersion = "latest"
+var currVersion = "local:HEAD"
+
+func init() {
+	if os.Getenv("PREV_ENGINE_VERSION") != "" {
+		prevVersion = os.Getenv("PREV_ENGINE_VERSION")
+	}
+
+	if os.Getenv("CURR_ENGINE_VERSION") != "" {
+		currVersion = os.Getenv("CURR_ENGINE_VERSION")
+	}
+}
+
+type GitbaseBackCompTestSuite struct {
+	cmdtest.RegressionSuite
+	timeout     time.Duration
+	testDir     string
+	testDataDir string
+}
+
+func TestGitbaseBackCompTestSuite(t *testing.T) {
+	s := GitbaseBackCompTestSuite{timeout: 10 * time.Minute}
+	suite.Run(t, &s)
+}
+
+func (s *GitbaseBackCompTestSuite) createRepo(name string) {
+	s.T().Helper()
+	require := s.Require()
+
+	repoPath := filepath.Join(s.testDir, name)
+	err := os.Mkdir(repoPath, os.ModePerm)
+	require.NoError(err)
+
+	cmd := exec.Command("git", "init", repoPath)
+	err = cmd.Run()
+	require.NoError(err)
+}
+
+func (s *GitbaseBackCompTestSuite) SetupSuite() {
+	token := os.Getenv("GITHUB_TOKEN")
+
+	// This is required because regression-core assumes that the working
+	// directory is the root of the project
+	s.setupWd()
+
+	config := regression.NewConfig()
+	cachePath := "regression-testing-cache"
+	config.BinaryCache = filepath.Join(cachePath, "binary")
+	config.RepositoriesCache = filepath.Join(cachePath, "repos")
+
+	tool := regression.Tool{
+		Name:        "engine",
+		BinaryName:  "srcd",
+		GitURL:      "https://github.com/src-d/engine",
+		ProjectPath: "github.com/src-d/engine",
+		BuildSteps: []regression.BuildStep{
+			{
+				Dir:     "",
+				Command: "make",
+				Args:    []string{"build", "docker-build"},
+			},
+		},
+	}
+
+	releases := regression.NewReleases("src-d", "engine", token)
+
+	s.PrevCmd = cmdtest.NewCommander(s.getBinary(config, tool, releases, prevVersion))
+	s.CurrCmd = cmdtest.NewCommander(s.getBinary(config, tool, releases, currVersion))
+}
+
+func (s *GitbaseBackCompTestSuite) setupWd() {
+	s.T().Helper()
+
+	_, filename, _, _ := runtime.Caller(0)
+	dir := path.Join(path.Dir(filename), "..", "..", "..")
+	err := os.Chdir(dir)
+	if err != nil {
+		s.T().Fatalf("cannot change working dir to %s: %v", dir, err)
+	}
+}
+
+func (s *GitbaseBackCompTestSuite) getBinary(conf regression.Config, tool regression.Tool, releases *regression.Releases, version string) string {
+	s.T().Helper()
+
+	bin := regression.NewBinary(conf, tool, version, releases)
+	if err := bin.Download(); err != nil {
+		s.T().Fatalf("%v", err)
+	}
+
+	return bin.Path
+}
+
+func (s *GitbaseBackCompTestSuite) SetupTest() {
+	testDir, err := ioutil.TempDir("", "gitbase-back-comp-test")
+	s.testDir = testDir
+	if err != nil {
+		s.T().Fatalf("%v", err)
+	}
+
+	homedir, err := homedir.Dir()
+	if err != nil {
+		s.T().Fatalf("unable to get home dir: %v", err)
+	}
+
+	s.testDataDir = filepath.ToSlash(filepath.Join(homedir, ".srcd", "gitbase"))
```

Comment:
let's add comment that we need it only for engine <0.12

Reply:
addressed [here](https://github.com/src-d/engine/pull/363/commits/864cd8a628be86ce8986bb50edae6fb52f48feca).